fvtest/compilertriltest/IficmpTest.cpp

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+#include <gtest/gtest.h>
+#include "Jit.hpp"
+#include "jitbuilder_compiler.hpp"
+
+#define ASSERT_NULL(pointer) ASSERT_EQ(nullptr, (pointer))
+#define ASSERT_NOTNULL(pointer) ASSERT_TRUE(nullptr != (pointer))
+
+extern "C" bool initializeJitWithOptions(char *options);
+
+using IfcmpReductionParamType = std::tuple<const char*, const char*, const char*, const char*>;
+
+class IfcmpReductionTest : public ::testing::TestWithParam<IfcmpReductionParamType>
+   {
+   public:
+
+   static void SetUpTestCase()
+      {
+      std::string defaultOptions = "-Xjit:acceptHugeMethods,enableBasicBlockHoisting,omitFramePointer,useIlValidator,paranoidoptcheck";
+      // Disable the following optimizations; otherwise, the trees are optimized into nothing.
+      std::string userOptions = "disableGlobalVP,disableLocalCSE";
+      std::string options = defaultOptions + "," + userOptions;
```

Comment:
Fixed. Not sure that the why is necessary; I assume that a curious reader would eyeball the tree and decide why for themselves.